src/golang/lib/engine/databricks_executor.go

```diff
@@ -0,0 +1,144 @@
+package engine
+
+import (
+	"context"
+	"time"
+
+	"github.com/aqueducthq/aqueduct/lib/collections/shared"
+	"github.com/aqueducthq/aqueduct/lib/job"
+	dag_utils "github.com/aqueducthq/aqueduct/lib/workflow/dag"
+	"github.com/aqueducthq/aqueduct/lib/workflow/operator"
+	"github.com/databricks/databricks-sdk-go/service/jobs"
+	"github.com/dropbox/godropbox/errors"
+	"github.com/sirupsen/logrus"
+)
+
+// We seperate out the execution step for Databricks Jobs since
+// Databricks takes care of launching Tasks.
+// Steps:
+// 1. Convert each operator into a Task (includes parent dependency).
+// 2. Create a multi-task Job with all the previously created Tasks.
+// 3. Launch job asynchronously
+// 4. Poll on each Task and update accordingly.
+func ExecuteDatabricks(
+	ctx context.Context,
+	workflowDag dag_utils.WorkflowDag,
+	workflowName string,
+	workflowRunMetadata *WorkflowRunMetadata,
+	timeConfig *AqueductTimeConfig,
+	opExecMode operator.ExecutionMode,
+	databricksJobManager *job.DatabricksJobManager,
+) error {
+	inProgressOps := workflowRunMetadata.InProgressOps
+	completedOps := workflowRunMetadata.CompletedOps
+	dag := workflowDag
+
+	// Convert the operators into tasks
+	taskList, err := CreateTaskList(ctx, dag, workflowName, databricksJobManager)
+	if err != nil {
+		return errors.Wrap(err, "Unable to convert operators to Databricks tasks.")
+	}
+
+	// Launch the workflow job with all tasks
+	_, err = databricksJobManager.LaunchMutlipleTaskJob(ctx, workflowName, taskList)
+	if err != nil {
+		return errors.Wrap(err, "Unable to launch workflow job on Databricks.")
+	}
+
+	for _, op := range dag.Operators() {
+		inProgressOps[op.ID()] = op
+	}
+
+	if len(inProgressOps) == 0 {
+		return errors.Newf("No initial operators to schedule.")
+	}
+
+	// Wait a little bit for all active operators to finish before exiting on failure.
+	defer waitForInProgressOperators(ctx, inProgressOps, timeConfig.OperatorPollInterval, timeConfig.CleanupTimeout)
+
+	start := time.Now()
+
+	for len(inProgressOps) > 0 {
+		if time.Since(start) > timeConfig.ExecTimeout {
+			return errors.New("Reached timeout waiting for workflow to complete.")
+		}
+
+		for _, op := range inProgressOps {
+			// Poll on the individual operator
+			execState, err := op.Poll(ctx)
+			logrus.Infof("execState of operator:%v", op.JobSpec().JobName())
+			logrus.Info(execState.Status)
+			if err != nil {
+				return err
+			}
+
+			if execState.Status == shared.PendingExecutionStatus {
+				continue
+			} else if execState.Status == shared.RunningExecutionStatus {
+				continue
+			}
+
+			if !execState.Terminated() {
```

Comment:
why the above two blocks couldn't be `if !execState.Terminated() { continue; }` ... and execute the rest without this check?

Reply:
Yep this is true, will change. I originally had some different (and unneeded) logic for a Pending state operator, but didn't edit out the different blocks.